lib/ProductOpener/Display.pm

```diff
@@ -1515,6 +1515,14 @@ sub can_use_query_cache() {
 			and (not $server_options{producers_platform}));
 }
 
+sub generate_query_cache_key ($name, $context_ref) {
+	# Generates a cache key taking the obsolete parameter into account
+	if (scalar param("obsolete")) {
```

Comment:
```suggestion
	if (scalar request_param("obsolete")) {
```

Just in case we have API search queries with the obsolete parameter in the JSON body in the future (we don't currently).